testing/integration/upgrade_fleet_test.go

```diff
@@ -426,3 +573,89 @@ func agentUpgradeDetailsString(a *kibana.AgentExisting) string {
 
 	return fmt.Sprintf("%#v", *a.UpgradeDetails)
 }
+
+// prepareFileServer prepares and returns a started HTTPS file server serving
+// files from rootDir and using cert as its TLS certificate.
+func prepareFileServer(t *testing.T, rootDir string, cert tls.Certificate) *httptest.Server {
+	// it's useful for debugging
+	dl, err := os.ReadDir(rootDir)
+	require.NoError(t, err)
+	var files []string
+	for _, d := range dl {
+		files = append(files, d.Name())
+	}
+	fmt.Printf("ArtifactsServer root dir %q, served files %q\n",
+		rootDir, files)
+
+	fs := http.FileServer(http.Dir(rootDir))
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Logf("[fileserver] %s - %s", r.Method, r.URL.Path)
+		fs.ServeHTTP(w, r)
+	}))
+
+	server.Listener, err = net.Listen("tcp", "127.0.0.1:443")
+	require.NoError(t, err, "could not create net listener for port 443")
+
+	server.TLS = &tls.Config{Certificates: []tls.Certificate{cert}}
+	server.StartTLS()
+	t.Logf("file server running on %s", server.URL)
+
+	return server
+}
+
+// prepareTLSCerts generates a CA and a child certificate for the given host and
```

Comment:
I am pretty sure there's already a function in test code that does this: can we avoid the duplication ?

Reply:
If the other function is only missing the hostname I would rather extend the existing function, generating certificates is tedious and error prone, I would rather avoid the duplication...

Not blocking, anyways